### ISSUE TITLE
return "foo" "bar";

### ISSUE THREAD

**Issue by fjf2002** [martin-pabst/Online-IDE-new-compiler#24]
The following program does not produce errors. The trailing "bar" is ignored, I think.

```java
String x() {
   return "foo" "bar";
}

println(x());
```

**Comment by Johannes-Pabst** [comment]
I've looked at the issue and found the following:

the core reason why your code doesn't throw an error is because the online-ide doesn't actually force you to write a semicolon after your return statement. `return ""` is just fine. Wheather this is necessarily bad can be debated.

Anyway, `return "foo" "bar";` isn't the only affected case because you can put a whole statement behind the return statement, e.g.:
`return "a" println("test");`
or even
`return "a" for(int i = 0; i < 10; i++){
         println("test");
 };`

the compiler just imagines a semicolon after the return statement and the `println("test");` is just the next statement, like this:
```
return "";
println("test");
```

As far as I can see, there are two options:
1. forcing students to put a semicolon after every return statements, so `return "";`, probably by modifying the `parseReturn` method in StatementParser.ts by adding a call to `expect(TokenType.semicolon)` if something is returned, like this:
```
parseReturn(): ASTReturnNode {
    let returnToken = this.getAndSkipToken();
    let term;
    if(this.tt==TokenType.semicolon){
        term=undefined;
    }else{
        term = this.parseTerm();
        this.expect(TokenType.semicolon);
    }
    while (this.comesToken(TokenType.semicolon, true)) { }
    return this.nodeFactory.buildReturnNode(returnToken, term);
}
```
or

2. allowing students to leave out unnecessary semicolons after `return`-statements while also allowing some unintuative code snippets

**Comment by martin-pabst** [comment]
Thank you!
(Johannes and I will learn how to make pull requests one day :-) )

**Comment by fjf2002** [comment]
@martin-pabst can the issue be closed?

**Comment by martin-pabst** [comment]
Yes, thank you!
